src/main/java/mate/academy/dao/BookDaoImpl.java

```diff
@@ -0,0 +1,119 @@
+package mate.academy.dao;
+
+import java.math.BigDecimal;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import mate.academy.exception.DataProcessingException;
+import mate.academy.lib.Dao;
+import mate.academy.model.Book;
+import mate.academy.service.ConnectionUtil;
+
+@Dao
+public class BookDaoImpl implements BookDao {
+
+    @Override
+    public Book create(Book book) {
+        String query = "INSERT INTO books (title, price) VALUES (?, ?)";
+
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement
+                        = connection.prepareStatement(query, Statement.RETURN_GENERATED_KEYS)) {
+            preparedStatement.setString(1, book.getTitle());
+            preparedStatement.setBigDecimal(2, book.getPrice());
+
+            int affectedRows = preparedStatement.executeUpdate();
+            if (affectedRows < 1) {
+                throw new RuntimeException("Expected to insert at least one row,"
+                        + " but inserted 0 rows");
```

Comment:
Instead of throwing a generic `RuntimeException`, consider using a more specific exception type or the existing `DataProcessingException` to maintain consistency in error handling.